compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -784,7 +784,8 @@ object SpaceEngine {
 
     def doShow(s: Space, flattenList: Boolean = false): String = s match {
       case Empty => "empty"
-      case Typ(c: ConstantType, _) => "" + c.value.value
+      case Typ(ConstantType(const : Constant), _) if const.tag == StringTag => "\"" + const.value + "\""
+      case Typ(ConstantType(const : Constant), _) => "" + const.value 
```

Comment:
I believe it can be replaced with:
```scala
case Typ(c: ConstantType, _) => c.show
```

Reply:
Almost, but it put me on the right track, thanks!